### ISSUE TITLE
Feat: Get API keys from cookie

### ISSUE THREAD

**Issue by gyopiazza** [metrico/qryn-view#161]
# Goal
A feature that allows to bypass the manual authentication in qryn-view and configure the instance URL.

# How
If the query param `cookie` is set in the URL, qryn-view should grab a cookie with the name set in the param, for example:
```
?cookie=some_cookie_name
```

When qryn-view detects the `cookie` query param it should try to get a cookie with the name `some_cookie_name` which contains the API key and secret exactly as expected by the proxy, for example:
```
SOME_LONG_API_KEY:SOME_LONG_API_SECRET
```
Note: this string doesn't need to be parsed and can be sent to the proxy as-is.

To configure the instance URL, we'll use the `url` query param, as follows:
```
?url=https://www.somedomain.com
```

# What
The query param name can be discussed and changed, but keeping generic would make it possible to release the feature on the open source repo (if necessary) without feeling out of context, since it makes its purpose very clear.

**Comment by jacovinus** [comment]
Some updates:
- param name for cookies will be 'cookie', for make it simple.
- UI will check for cookie name inside cookies and retireve information
- We need to get (at least) an API URL, and (basic) auth if there is one needed, for filling data sources data and then make the request.

**Comment by gyopiazza** [comment]
@jacovinus I've updated the task as discussed, please feel free to propose different names for the query params since this is domain of qryn-view 😊

Thank you!

**Comment by jacovinus** [comment]
we could use something less descriptive if the base API url is the self url, as ?url=self

**Comment by gyopiazza** [comment]
@jacovinus what do you think if we do it like this?
- If there's no `url` query param => use the current url
- If there's a `url` query param => use it as the url

FYI @oriolcastro

**Comment by oriolcastro** [comment]
If what controls the _"auto-configuration of the data sources"_ is the presence of the cookie search param then it's ok.

Otherwise, I see a potential issue for non-Gigapipe users

**Comment by jacovinus** [comment]
> If what controls the _"auto-configuration of the data sources"_ is the presence of the cookie search param then it's ok.
> 
> Otherwise, I see a potential issue for non-Gigapipe users
It was already taken in consideration. Final one should be:
 - If there's cookie and url, take key:secret from cookie and url from url param
 - If there's cookie and no url, take base url
 - If there's no cookie and no url, take data sources settings
 - If there's no cookie and is url, take that url for all datasources (no auth required case)

**Comment by jacovinus** [comment]
@oriolcastro , this feature is available at dev branch, you can pull and test it at local or try it with vercel
If everything ok, then we close this issue.

**Comment by jacovinus** [comment]
solved at [985babe](https://github.com/metrico/qryn-view/commit/985babe4fc12af6c208d5d89a5f62e2cc81d435d) and tested, closing.
